osu.Game/Database/StandardisedScoreMigrationTools.cs

```diff
@@ -26,7 +26,7 @@ public static bool ShouldMigrateToNewStandardised(ScoreInfo score)
             if (score.IsLegacyScore)
                 return false;
 
-            if (score.TotalScoreVersion > 30000004)
+            if (score.TotalScoreVersion > 30000005)
```

Comment:
Tempted to also make this `>= LegacyScoreDecoder.LATEST_VERSION` at this point, but I also am deeply confused by the legacy score version mechanism at this point. @smoogipoo please advise.

Reply:
> What I mean is, is there ever a chance this could run before scores are upgraded to LATEST_VERSION?

Can you elaborate? I'm not sure I follow...

Reply:
Let's say this changed line was merged in, and a score exists in the database with version 005.

1. This condition now fails for that score.
2. We compare `GetOldStandardised(score) == score`. This evaluates false.
3. Likewise `scoreIsVeryOld` will also evaluate false.
4. The method returns `false` and we'll "upgrade" the score via `GetNewStandardised()`.

As I understand it, the above upgrade is lossy. I would prefer that we keep current scorev2-esque scores the way they are so this upgrade isn't done.

_If_ the 005 score would first get upgraded to 006 via your change to the `BackgroundDataStoreProcessor`, that would cause the condition in (1) to now pass, then this change is fine.

Reply:
> 4\. The method returns `false` and we'll "upgrade" the score via `GetNewStandardised()`

Um... If `ShouldMigrateToNewStandardised()` returns false, which as you said it will, then the score will not be upgraded...?

https://github.com/ppy/osu/blob/469a659938dfb3c696c9a9fb51c7252214014a31/osu.Game/Scoring/ScoreImporter.cs#L105-L111
https://github.com/ppy/osu/blob/469a659938dfb3c696c9a9fb51c7252214014a31/osu.Game/Database/RealmAccess.cs#L950-L960

On the other side, scores imported from stable receive version 30000001:

https://github.com/ppy/osu/blob/469a659938dfb3c696c9a9fb51c7252214014a31/osu.Game/Scoring/Legacy/LegacyScoreDecoder.cs#L52-L57

so the `BackgroundDataStoreProcessor` change was supposed to make them reprocess...

That said I'll go recheck empirically again because this stuff is very confusing (as it was every time up to now).

Reply:
Oh sorry, I mentally inverted the condition of `oldScoreMatchesExpectations`. Probably worth going over this again with a fine comb, because it's very hard to understand.

Reply:
Looking at this with a finer comb again,

> Afaik this was only here to convert scores <= 002 or so (old, linear-style standardised scoring) to the new ScoreV2-style one.

I'm pretty sure you are correct here and this was wrongly bumped in 5dee43815c2051cd2bae237082f0307f68e1a283, with the thought that it was going to lead to scores getting reprocessed, while it was `BackgroundDataStoreProcessor` that was supposed to be changed.

I've reverted this in https://github.com/ppy/osu/pull/25876/commits/ddb67c87a8c20680c6ef77d8faa512b50d090be5, but I'd probably want to double-check with @peppy that that is what happened there...

Reply:
Everything said here looks to check out.